### PR TITLE
MAINT: Further small return value validation fixes

### PR DESCRIPTION
**Pull request by seberg**
Half of these were probably just multiple versions of the same pattern, so it got lost.

### CODE REVIEW

File: numpy/core/src/multiarray/buffer.c [hunk 1]
```diff
@@ -1057,7 +1057,7 @@ _descriptor_from_pep3118_format_fast(char const *s, PyObject **result)
     else {
         *result = (PyObject*)PyArray_DescrNewByteorder(descr, byte_order);
         Py_DECREF(descr);
-        if (result == NULL) {
+        if (*result == NULL) {
```
Comment (seberg) [inline numpy/core/src/multiarray/buffer.c:1060]: Well, clang is smart enough to notice that it was always false, to bad GCC running in CI is not :)

**Comment by seberg** [comment]
Ping @charris, can you smuggle this into the release you are currently preparing?

File: numpy/core/src/multiarray/descriptor.c [hunk 2]
```diff
@@ -597,9 +597,7 @@ _convert_from_array_descr(PyObject *obj, int align)
 
     PyArray_Descr *new = PyArray_DescrNewFromType(NPY_VOID);
     if (new == NULL) {
-        Py_XDECREF(fields);
-        Py_XDECREF(nameslist);
-        return NULL;
+        goto fail;
```
Comment (seberg) [inline numpy/core/src/multiarray/descriptor.c:600]: I had not realized we had a goto here that could be used.

File: tools/swig/pyfragments.swg [hunk 3]
```diff
@@ -52,7 +52,7 @@
     }
 %#endif
     if (!PyArray_IsScalar(obj,Integer)) return SWIG_TypeError;
-    PyArray_Descr * longDescr = PyArray_DescrNewFromType(NPY_LONG);
+    PyArray_Descr * longDescr = PyArray_DescrFromType(NPY_LONG);
```
Comment (seberg) [inline tools/swig/pyfragments.swg:55]: `PyArray_DescrFromType` cannot fail for builtin types (except char)... No idea if it matters, but this is fine.

**Comment by charris** [comment]
> can you smuggle this into the release you are currently preparing?

Yes.

File: tools/swig/pyfragments.swg (hunk 3 above)
Comment (charris) [reply tools/swig/pyfragments.swg:55]: Makes me wonder about the use of `PyArray_DescrNewFromType`. AFAICT, almost all uses are with builtin types.

Comment (charris) [reply tools/swig/pyfragments.swg:55]: There are about 14 uses for types that aren't mutable. Looks like an opportunity for some cleanup.

**Comment by charris** [comment]
Thanks Sebastian.

**Pull request merged**